src/NSwag.CodeGeneration/ClientGeneratorBase.cs

```diff
@@ -160,6 +161,9 @@ private List<TOperationModel> GetOperations(OpenApiDocument document)
 
                     var operationName = BaseSettings.OperationNameGenerator.GetOperationName(document, path, httpMethod, operation);
 
+                    // Strip any character that is not a word character or a period
+                    operationName = Regex.Replace(operationName, @"[^\w\.-]", "", RegexOptions.None);
```

Comment:
performance will be probably bad as your are calling dynamic regex inside a loop

Reply:
@lahma I'm not sure if that really makes a difference, but the RegexOptions.Compiled could be used to speed it up. I can see that dynamic regexe's are used in a few other places though (like `SingleClientFromPathSegmentsOperationNameGenerator`).

I'm not facing this issue with `MultipleClientsFromPathSegments`, but I would really like to generate a single client. Maybe it just involves using similar logic in the other generators to avoid invalid names?

Reply:
Okay I just found out that the curly braces are actually handled fine when using `SingleClientFromPathSegments`. I had a single area called `/upload?id={id}` that caused it to generate incorrect methods names.